### PR TITLE
service, loadbalancer: Misc refactoring

### PR DESCRIPTION
**Pull request by brb**
Please see commit msgs.

### CODE REVIEW

**Review by vadorovsky — APPROVED** [review]

**Comment by brb** [comment]
test-me-please

File: pkg/maps/lbmap/lbmap.go [hunk 1]
```diff
@@ -370,7 +370,13 @@ func updateMasterService(fe ServiceKey, nbackends int, revNATID int, svcType loa
 	zeroValue := fe.NewValue().(ServiceValue)
 	zeroValue.SetCount(nbackends)
 	zeroValue.SetRevNat(revNATID)
-	zeroValue.SetFlags(loadbalancer.CreateSvcFlag(svcLocal, sessionAffinity, !fe.IsSurrogate(), svcType).UInt8())
+	flag := loadbalancer.NewSvcFlag(&loadbalancer.SvcFlagParam{
```
Comment (sayboras) [inline pkg/maps/lbmap/lbmap.go:373]: unrelated to this PR, I feel like this method can be changed to take struct argument as well https://github.com/cilium/cilium/blob/4f73469c9e32f5ca0e47670fe443272366cc5ad5/pkg/maps/lbmap/lbmap.go#L49

**Review by sayboras — APPROVED** [review]
LGTM

File: pkg/maps/lbmap/lbmap.go (hunk 1 above)
Comment (brb) [reply pkg/maps/lbmap/lbmap.go:373]: Yeah, it's coming. I didn't want to touch the lbmap pkg in this PR.

Comment (sayboras) [reply pkg/maps/lbmap/lbmap.go:373]: nice :100:, sorry to jump in too soon

**Review by christarazi — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by brb** [comment]
Added `needs-backport/1.8`, as the LB src range check depends on it.

**Pull request merged**